hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java

```diff
@@ -53,41 +62,100 @@ public ScheduledTaskDescriptor() {
     public ScheduledTaskDescriptor(TaskDefinition definition) {
         this.definition = definition;
         this.state = new HashMap();
-        this.stats.set(new ScheduledTaskStatisticsImpl());
+        this.stats = new ScheduledTaskStatisticsImpl();
     }
+
     public ScheduledTaskDescriptor(TaskDefinition definition,
-                                   Map<?, ?> state, ScheduledTaskStatisticsImpl stats) {
+                                   Map<?, ?> state, ScheduledTaskStatisticsImpl stats,
+                                   ScheduledTaskResult result) {
         this.definition = definition;
-        this.stats.set(stats);
+        this.stats = stats;
         this.state = state;
+        this.resultRef.set(result);
     }
 
     public TaskDefinition getDefinition() {
         return definition;
     }
 
     ScheduledTaskStatisticsImpl getStatsSnapshot() {
-        return stats.get().snapshot();
+        return stats.snapshot();
     }
 
-    void setStats(ScheduledTaskStatisticsImpl stats) {
-        this.stats.set(stats);
+    Map<?, ?> getStateSnapshot() {
+        return state;
     }
 
-    Map<?, ?> getStateSnapshot() {
-        return new HashMap(state);
+    ScheduledTaskResult getTaskResult() {
+        return resultRef.get();
+    }
+
+    void setStats(ScheduledTaskStatisticsImpl stats) {
+        this.stats = stats;
     }
 
     void setState(Map<?, ?> snapshot) {
         this.state = snapshot;
     }
 
     ScheduledFuture<?> getScheduledFuture() {
-        return scheduledFuture;
+        return future;
     }
 
     void setScheduledFuture(ScheduledFuture<?> future) {
-        this.scheduledFuture = future;
+        this.future = future;
+    }
+
+    void setTaskResult(ScheduledTaskResult result) {
+        this.resultRef.set(result);
+    }
+
+    Object get()
+            throws ExecutionException, InterruptedException {
+
+        ScheduledTaskResult result = resultRef.get();
```

Comment:
Should we also check if this we already have a result here in `resultRef` (this was a one-time callable which has migrated to a different node)?